Accept non-standard bools in more CUDA kernels

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78957

This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`load_vector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

peterbell10 committed:
Accept non-standard bools in more CUDA kernels

This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`LoadVector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

[ghstack-poisoned]

peterbell10 committed:
Update on "Accept non-standard bools in more CUDA kernels"


This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`LoadVector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

[ghstack-poisoned]

peterbell10 committed:
Update on "Accept non-standard bools in more CUDA kernels"


This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`LoadVector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -11220,7 +11217,8 @@ def error_inputs_mean(op_info, device, **kwargs):
            supports_fwgrad_bwgrad=True,
            skips=(
-               DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_non_standard_bool_values'),
+               DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_non_standard_bool_values',
```
Nice

mruberry commented:
I added the ciflow/trunk label to trigger more tests, just in case, since this is modifying the OpInfos

peterbell10 committed:
Replace LoadVector struct with a function on "Accept non-standard bools in more CUDA kernels"


This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`LoadVector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

[ghstack-poisoned]

peterbell10 committed:
Update on "Accept non-standard bools in more CUDA kernels"


This fixes all remaining CUDA kernels, except those using `cub` or
`thrust`, to accept boolean tensors with values other than 1 or 0.

I do this by using `c10::load` in more places, and also adding a
`load_vector` helper into `MemoryAccess.cuh` that does the same thing
for vectorized loads.

[ghstack-poisoned]

peterbell10 commented:
@mruberry PTAL

mruberry reviewed (approved):
Thanks @peterbell10!

mruberry commented:
@pytorchbot merge
